src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -32,32 +74,250 @@ const NotebookLauncher = _.flow(
     title: _.get('notebookName'),
     showTabBar: false
   })
-)(({ queryParams = {}, notebookName, workspace, workspace: { accessLevel, canCompute }, cluster, refreshClusters, onRequesterPaysError }, ref) => {
-  return (Utils.canWrite(accessLevel) && canCompute && !queryParams['read-only']) ?
-    h(NotebookEditor, { notebookName, workspace, cluster, refreshClusters }) :
-    h(Fragment, [
-      h(ReadOnlyMessage, { notebookName, workspace }),
-      h(NotebookPreviewFrame, { notebookName, workspace, onRequesterPaysError })
+)(
+  ({ queryParams = {}, notebookName, workspace, workspace: { workspace: { namespace, name }, accessLevel, canCompute }, cluster, refreshClusters, onRequesterPaysError }, ref) => {
+    const [createOpen, setCreateOpen] = useState(false)
+    // Status note: undefined means still loading, null means no cluster
+    const clusterStatus = cluster && cluster.status
+
+    return h(Fragment, [
+      Utils.cond([
+        !(Utils.canWrite(accessLevel) && canCompute), () => h(Fragment, [
+          h(PreviewHeader, { cluster, refreshClusters, notebookName, workspace, readOnlyAccess: true }),
+          h(NotebookPreviewFrame, { notebookName, workspace })
+        ])
+      ], [
+        Utils.canWrite(accessLevel) && canCompute && queryParams['edit'] && clusterStatus === 'Running',
+        () => h(NotebookEditorFrame, { key: cluster.clusterName, workspace, cluster, notebookName, mode: 'Edit' })
+      ], [
+        Utils.canWrite(accessLevel) && canCompute && queryParams['playground'] && clusterStatus === 'Running',
+        () => h(NotebookEditorFrame, { key: cluster.clusterName, workspace, cluster, notebookName, mode: 'Playground' })
+      ],
+      () => h(Fragment, [
+        h(PreviewHeader, { cluster, refreshClusters, notebookName, workspace, readOnlyAccess: false }),
+        h(NotebookPreviewFrame, { notebookName, workspace })
+      ])),
+      (queryParams.edit || queryParams.playground) && h(ClusterKicker, { cluster, refreshClusters, onNullCluster: () => { setCreateOpen(true) } }),
+      createOpen && h(NewClusterModal, {
```

Comment:
It seems strange to have two possible instances of this modal on the page. Maybe we could combine them by moving some state around.